plan/logical_plan_builder.go

```diff
@@ -589,6 +589,7 @@ func (b *planBuilder) buildProjection4Union(u *LogicalUnionAll) {
 			srcType := srcCol.RetType
 			if !srcType.Equal(dstType) {
 				exprs[i] = expression.BuildCastFunction(b.ctx, srcCol.Clone(), dstType)
+				exprs[i].GetType().Flag |= mysql.ParseToJSONFlag
```

Comment:
This is a bug fix?  Please add a test case for it.

Reply:
Added in `write_test.go`.

Reply:
For `union` it's hard to test because `dstType` == `JSON` will never happens.